test/integration/good/compiler-optimizations/cpp.expected

```diff
@@ -26161,7 +26219,7 @@ class optimizations_model final : public model_base_crtp<optimizations_model> {
         dataonlyvar = 3;
         current_statement__ = 85;
         lp_accum__.add(dataonlyvar);
-        local_scalar_t__ paramvar;
+        local_scalar_t__ paramvar = DUMMY_VAR__;
```

Comment:
Why are all these test results gaining initializers that aren't present before?

Reply:
So the associated code for this is 

```
    real paramvar = 3;
    if (42 > 1)
      paramvar = theta * 34;
    target += paramvar;
```

I think before I was being less conservative and looking into blocks for the next assignment as well. The compiler would get rid of the if statement here since it's always true and generate just a block, which the optimizer looked through and said we don't need to initialize it. 

```cpp
        local_scalar_t__ paramvar;
        {
          current_statement__ = 87;
          paramvar = (theta * 34);
        }
```

But I think there can be cases we will have false positives with this so I'm more conservative now and only look if the immediate next line is an assignment to the declared variable.